Cave.java

```diff
@@ -1,6 +1,5 @@
-class Cave extends Terrain {
+class Cave extends EarthTerrain {
```

Comment:
Did you mean "CaveTerrain"?
Notice that your file name should match your class name.